puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/release/foreman-installer-develop-release.groovy

```diff
@@ -0,0 +1,83 @@
+def commit_hash = ''
+def sourcefile = ''
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+    }
+
+    environment {
+        ruby_version = "2.4"
+    }
+
+    stages {
+        stage('Setup Environment') {
+            steps {
+                dir('foreman-installer') {
+                    git(url: 'https://github.com/theforeman/foreman-installer.git', branch: 'develop')
+                    script { commit_hash = sh(script: "git rev-parse HEAD | tee commit", returnStdout: true).trim() }
+                    archiveArtifacts(artifacts: 'commit')
+                }
+                dir('foreman-packaging') { git(url: 'https://github.com/theforeman/foreman-packaging.git', branch: 'rpm/develop') }
+                setup_obal()
+                configureRVM(env.ruby_version)
+            }
+        }
+        stage('Build and Archive Source') {
+            steps {
+                dir('foreman-installer') {
+                    withRVM(["bundle install --jobs 5 --retry 5"], env.ruby_version)
+                    withRVM(["bundle exec rake pkg:generate_source"], env.ruby_version)
+                    archiveArtifacts(artifacts: 'pkg/*')
+                    dir('pkg') {
+                        script { sourcefile = sh(script: "ls", returnStdout: true).trim() } // i could probably do this with a better built-in
```

Comment:
Seems fine, I think you can drop the comment

Reply:
What if there are multiple files? It'd be safer to delete the `pkg` directory before calling `pkg:generate_source`. Even though the `deleteDir()` is in the `always()` block, I wouldn't count on it 100% because a slave might disconnect during the build and I don't know what happens then.

Reply:
I like this. In https://github.com/theforeman/foreman/commit/bdbbecb9cfbead3582b030070a8a37025f5dea6a#diff-442dfed27b160ff7165f10d1d0ba9630 I modified it to print the filename while generating but a dedicated task is cleaner and reusable.